sdks/python/apache_beam/runners/worker/data_plane.py

```diff
@@ -129,15 +128,19 @@ def create(
 class SizeBasedBufferingClosableOutputStream(ClosableOutputStream):
   """A size-based buffering OutputStream."""
 
+  _large_flush_last_observed_timestamp = 0.0
+
   def __init__(
       self,
       close_callback=None,  # type: Optional[Callable[[bytes], None]]
       flush_callback=None,  # type: Optional[Callable[[bytes], None]]
-      size_flush_threshold=_DEFAULT_SIZE_FLUSH_THRESHOLD  # type: int
+      size_flush_threshold=_DEFAULT_SIZE_FLUSH_THRESHOLD,  # type: int
+      large_buffer_warn_threshold_mib = 512  # type: int
```

Comment:
nit: I'd just let this be in bytes (for consistency with everything else) rather than have `_mib` and do the conversion elsewhere.

Reply:
ok, will send a follow up cleanup PR to make sure this catches the release